src/ansiblelint/rules/MissingFilePermissionsRule.py

```diff
@@ -33,6 +34,63 @@
     'template',
 )
 
+_FQCN_MODS_MAP: List[Tuple[str, List[str]]] = [
+    (
+        # fqcn prefix:
+        'ansible.builtin',
+        # modules:
+        [
+            # _modules:
+            'assemble',
+            'copy',
+            'file',
+            'replace',
+            'template',
+            # _modules_with_create:
+            'blockinfile',
+            'lineinfile',
+        ],
+    ),
+    (
+        'community.general',
+        [
+            # _modules:
+            'archive',
+            # _modules_with_create:
+            'htpasswd',
+            'ini_file',
+        ],
+    ),
+]
+_MOD_FQCN_PREFIX_MAP: Dict[str, str] = dict(
+    itertools.chain.from_iterable(
+        ((mod, prefix) for mod in modules) for prefix, modules in _FQCN_MODS_MAP
+    )
+)
+
+_MODULES = {
+    'archive',
+    'assemble',
+    'copy',  # supports preserve
+    'file',
+    'replace',  # implicit preserve behavior but mode: preserve is invalid
+    'template',  # supports preserve
+    # 'unarchive',  # disabled because .tar.gz files can have permissions inside
+}
+for mod in _MODULES.copy():
+    _MODULES.add(f'{_MOD_FQCN_PREFIX_MAP[mod]}.{mod}')
+
+_MODULES_WITH_CREATE = {
+    'blockinfile': False,
+    'htpasswd': True,
+    'ini_file': True,
+    'lineinfile': False,
+}
+for mod in _MODULES_WITH_CREATE.copy():
+    _MODULES_WITH_CREATE[f'{_MOD_FQCN_PREFIX_MAP[mod]}.{mod}'] = _MODULES_WITH_CREATE[
+        mod
+    ]
```

Comment:
The same thing here. `_MODULES_WITH_CREATE` should be a plain dictionary with all options directly listed.